### PR TITLE
Babysitter

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

**Comment by andrewconner** [comment]
@2is10 Please review, thanks.

File: shoebox/app/com/keepit/common/healthcheck/Babysitter.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@ImplementedBy(classOf[BabysitterImpl])
+trait Babysitter {
+  def watch[A](warnTimeout: akka.util.FiniteDuration, errorTimeout: akka.util.FiniteDuration)(block: => A)(implicit app: Application): A
+}
+
+class BabysitterImpl extends Babysitter with Logging {
+  def watch[A](warnTimeout: akka.util.FiniteDuration, errorTimeout: akka.util.FiniteDuration)(block: => A)(implicit app: Application): A = {
+    val startTime = inject[DateTime]
+    val e = new Exception("Babysitter error timeout")
+    val babysitter = Akka.system.scheduler.scheduleOnce(errorTimeout) {
```
Comment (2is10) [inline shoebox/app/com/keepit/common/healthcheck/Babysitter.scala]: as discussed, let's inject a fake scheduler here, so you can run the block immediately and assert the desired result

File: shoebox/test/com/keepit/common/healthcheck/BabysitterTest.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        val now = new DateTime(2012, 5, 31, 4, 3, 2, 1, DEFAULT_DATE_TIME_ZONE)
+        // Babysitter gets time twice. At the beginning, and at the end for warnTimeout
+        inject[FakeClock].push(now)
+        inject[FakeClock].push(now.minusSeconds(5))
+
+        inject[Babysitter].watch(Duration(1, "seconds"), Duration(1, "seconds")) {
+          // So slow!
+        }
+
+        1===1
```
Comment (2is10) [inline shoebox/test/com/keepit/common/healthcheck/BabysitterTest.scala:42]: Shouldn't this be:  inject[HealthcheckPlugin].errorCount === 1

File: shoebox/test/com/keepit/common/healthcheck/BabysitterTest.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  "Babysitter" should {
+    "do nothing if code executes quickly" in {
+      running(new EmptyApplication().withFakeHealthcheck().withFakeTime().withRealBabysitter().withFakeScheduler()) {
+
+        inject[Babysitter].watch(Duration(1, "seconds"), Duration(1, "seconds")) {
+          // So fast!
+        }
+
+        inject[HealthcheckPlugin].errorCount == 0
```
Comment (2is10) [inline shoebox/test/com/keepit/common/healthcheck/BabysitterTest.scala:31]: Should this be a three-equals?  === 0

File: shoebox/test/com/keepit/test/TestApplication.scala [hunk 4]
```diff
@@ -57,3 +83,40 @@ class FakeClock {
...
+    fakeCancellable
+  }
+
+  def schedule(initialDelay: Duration,frequency: Duration,receiver: ActorRef,message: Any): Cancellable = fakeCancellable
+  def schedule(initialDelay: Duration, frequency: Duration)(f: => Unit): Cancellable = immediateExecutor(f)
+  def schedule(initialDelay: Duration, frequency: Duration, runnable: Runnable): Cancellable = fakeCancellable
+  def scheduleOnce(delay: Duration, receiver: ActorRef, message: Any): Cancellable = fakeCancellable
+  def scheduleOnce(delay: Duration)(f: ⇒ Unit): Cancellable = immediateExecutor(f)
+  def scheduleOnce(delay: Duration, runnable: Runnable): Cancellable = fakeCancellable
+}
```
Comment (2is10) [inline shoebox/test/com/keepit/test/TestApplication.scala:121]: Nice fake!

**Pull request merged**